### PR TITLE
Added possibility to give labels to aircraft

### PR DESCRIPTION
**Pull request by matejglejtek**

### CODE REVIEW

**Comment by marianhlavac** [comment]
```
❯ flutter run
This app is using a deprecated version of the Android embedding.
To avoid unexpected runtime failures, or future build failures, try to migrate this app to the V2 embedding.
Take a look at the docs for migrating an app: https://github.com/flutter/flutter/wiki/Upgrading-pre-1.12-Android-projects
The plugin `device_info_plus` requires your app to be migrated to the Android embedding v2. Follow the steps on https://flutter.dev/go/android-project-migration and re-run this
command.
```

Budeme muset zmigrovat projekt

File: lib/bloc/aircraft/aircraft_cubit.dart [hunk 1]
```diff
@@ -178,6 +189,48 @@ class AircraftCubit extends Cubit<AircraftState> {
     }
   }
 
+  // labels are stored locally on device,
```
Comment (marianhlavac) [inline lib/bloc/aircraft/aircraft_cubit.dart:192]: It's better to describe what the method does, so something more like "Retrieves the labels stored persistently locally on the device". Same goes for the `addAircraftLabel` method, better to write "Stores the label persistently locally on the device"

File: lib/bloc/map/map_cubit.dart [hunk 2]
```diff
@@ -275,29 +275,43 @@ class MapCubit extends Cubit<GMapState> {
...
                     markerHue = gmap.BitmapDescriptor.hueBlue;
                   }
-
+                  var haslocation = true;
+                  if (e.isEmpty ||
+                      e.last.locationMessage == null ||
+                      e.last.locationMessage?.latitude == null ||
+                      e.last.locationMessage?.longitude == null) {
+                    haslocation = false;
+                  }
```
Comment (marianhlavac) [inline lib/bloc/map/map_cubit.dart:284]:
```suggestion
                  final haslocation = !(e.isEmpty ||
                      e.last.locationMessage == null ||
                      e.last.locationMessage?.latitude == null ||
                      e.last.locationMessage?.longitude == null);
```

A bit unnecessary variable modification and if condition here

File: lib/widgets/sliders/aircraft/aircraft_card_title.dart [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+                  size: Sizes.textIconSize, color: Colors.black),
+            ),
+            TextSpan(
+              text: ' $givenLabel',
+            ),
+          ],
+        ),
+      );
+    }
+  }
```
Comment (marianhlavac) [inline lib/widgets/sliders/aircraft/aircraft_card_title.dart:57]:
```suggestion
      return Text.rich(
        TextSpan(
          style: const TextStyle(
            fontWeight: FontWeight.bold,
          ),
          children: [
          if (givenLabel != null)
            WidgetSpan(
                alignment: PlaceholderAlignment.middle,
                child: Icon(Icons.label_outline,
                    size: Sizes.textIconSize, color: Colors.black),
              )
          else if (uasId.startsWith('1596') == true)
              WidgetSpan(
                alignment: PlaceholderAlignment.middle,
                child: Image.asset('assets/images/dronetag.png',
                    height: 16,
                    width: 24,
                    alignment: Alignment.topRight,
                    color: Colors.black),
              ),
            TextSpan(
              text: givenLabel ?? uasId,
            ),
          ],
        ),
      );
```

Some parts of the code here needs not to be duplicated

**Review by marianhlavac — CHANGES_REQUESTED** [review]
Awesome feature 😄 Good job. I have just a few remarks.

File: lib/widgets/sliders/aircraft/detail/aircraft_label_text.dart [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+                padding: const EdgeInsets.all(1.0),
+                iconSize: 20,
+                icon: const Icon(
+                  Icons.delete,
+                  color: Colors.white,
+                ),
+                onPressed: () {
+                  showAlertDialog(
+                    context,
+                    'Would you really like to delete aircraft label?',
```
[the remark this answers is not in the document]
Comment (matejglejtek) [reply lib/widgets/sliders/aircraft/detail/aircraft_label_text.dart:119]: changed according to suggestion

File: lib/widgets/sliders/aircraft/aircraft_card_title.dart (hunk 3 above)
Comment (matejglejtek) [reply lib/widgets/sliders/aircraft/aircraft_card_title.dart]: duplicates removed

File: lib/bloc/map/map_cubit.dart (hunk 2 above)
Comment (matejglejtek) [reply lib/bloc/map/map_cubit.dart]: suggestion adopted

**Review by marianhlavac — APPROVED** [review]
Looks good, please resolve the conflicts that has appeared and it's ready to merge.

**Pull request merged**